R/summary_funs.R

```diff
@@ -202,198 +209,223 @@ weighted_summary_means <- function(y_wobs_test, family, wdraws, mu, dis, cl_ref,
-## Currently, these `wcv` are nonconstant (and not `NULL`) only in case of
-## subsampled PSIS-LOO CV. The actual observation weights (specified by the
+## The actual observation weights (specified by the
 ## user) are contained in `y_wobs_test$wobs`. These are already taken into
 ## account by `<refmodel_object>$family$ll_fun()` (or
 ## `<refmodel_object>$family$latent_ll_oscale()`) and are thus already taken
 ## into account in `lppd`. However, `mu` does not take them into account, so
 ## some further adjustments are necessary below.
-get_stat <- function(mu, lppd, y_wobs_test, stat, mu.bs = NULL, lppd.bs = NULL,
-                     wcv = NULL, alpha = 0.1, ...) {
-  n_notna.bs <- NULL
-  if (stat %in% c("elpd", "mlpd", "gmpd")) {
-    if (!is.null(lppd.bs)) {
-      # Compute the performance statistics using only those observations for
-      # which both `lppd` and `lppd.bs` are not `NA`:
-      lppd[is.na(lppd.bs)] <- NA
-      lppd.bs[is.na(lppd)] <- NA
-      n_notna.bs <- sum(!is.na(lppd.bs))
-    }
-    n_notna <- sum(!is.na(lppd))
-    n <- length(lppd)
-  } else {
-    hasNA_y <- is.na(y_wobs_test$y_prop %||% y_wobs_test$y)
-    if (!is.null(mu.bs)) {
-      # Compute the performance statistics using only those observations for
-      # which both `mu` and `mu.bs` are not `NA`:
-      mu[is.na(mu.bs)] <- NA
-      mu.bs[is.na(mu)] <- NA
-      n_notna.bs <- sum(!is.na(mu.bs) & !hasNA_y)
-    }
-    n_notna <- sum(!is.na(mu) & !hasNA_y)
-    n <- length(mu)
-  }
-  if (!is.null(n_notna.bs) && getOption("projpred.additional_checks", FALSE)) {
-    stopifnot(n_notna == n_notna.bs)
-  }
-  if (n_notna == 0) {
+get_stat <- function(summaries, summaries_baseline = NULL,
+                     summaries_fast = NULL,
+                     y_wobs_test, stat, alpha = 0.1, ...) {
+  mu <- summaries$mu
+  lppd <- summaries$lppd
+  loo_inds <- which(!is.na(lppd))
+  n <- length(lppd)
+  n_loo <- length(loo_inds)
+  ## is this needed anymore?
+  ## n_notna.bs <- NULL
+  ## if (!is.null(summaries_fast)) {
+  ##   # Compute the performance statistics using only those observations for
+  ##   # which fast summaries are not NA
+  ##   if (stat %in% c("elpd", "mlpd", "gmpd")) {
+  ##     lppd[is.na(lppd_baseline)] <- NA
+  ##     n_notna.bs <- sum(!is.na(lppd_baseline))
+  ##   }
+  ##   n_notna <- sum(!is.na(lppd))
+  ##   n <- length(lppd)
+  ## } else {
+  ##   hasNA_y <- is.na(y_wobs_test$y_prop %||% y_wobs_test$y)
+  ##   if (!is.null(mu_baseline)) {
+  ##     # Compute the performance statistics using only those observations for
+  ##     # which both `mu` and `mu_baseline` are not `NA`:
+  ##     mu[is.na(mu_baseline)] <- NA
+  ##     mu_baseline[is.na(mu)] <- NA
+  ##     n_notna.bs <- sum(!is.na(mu_baseline) & !hasNA_y)
+  ##   }
+  ##   n_notna <- sum(!is.na(mu) & !hasNA_y)
+  ##   n <- length(mu)
+  ## }
+  ## if (!is.null(n_notna.bs) && getOption("projpred.additional_checks", FALSE)) {
+  ##   stopifnot(n_notna == n_notna.bs)
+  ## }
```

Comment:
Currently, I think this can indeed be removed. At first, I was worried about the augmented-data projection (or the latent projection with a `factor`-like response and performance evaluation on original response scale) because there, I thought `mu` could be of length `n_obs * n_cats` (`n_obs`: number of observations, `n_cats`: number of response categories), but it turns out that this is never the case: The only permitted performance statistic where `mu` is relevant in those cases is `"acc"` (`== "pctcorr"`). And there, due to the application of `catmaxprb()` before getting into `get_stat()`, object `mu` has indeed always length `n_obs` (just like `lppd`). So everything should be fine there. I was also worried about the case where `y` is all-`NA`s. This occurs for example when using the latent projection in combination with K-fold CV and `resp_oscale = FALSE`. However, in that case, `lppd` here in `get_stat()` should also be all-`NA`s because `y` is used for computing `lppd`. So everything should be fine there as well.

Reply:
removed